### PR TITLE
feat: add custom coder bin path for ProxyCommand

### LINKED ISSUE #8217: environment variable for `coder config-ssh` to specify path and binary for CLI
A customer would like to pass an environment variable to `coder config-ssh` to specify the path and binary of the CLI in each `config` entry's `ProxyCommand`

e.g., `coder config-ssh --cli-path`

### PR DESCRIPTION
**Pull request by Emyrk**
Closes https://github.com/coder/coder/issues/8217

### CODE REVIEW

File: cli/configssh.go [hunk 1]
```diff
@@ -233,10 +234,16 @@ func (r *RootCmd) configSSH() *clibase.Cmd {
...
-				return err
+
+			var err error
+			coderBinary := coderCliPath
+			if coderBinary == "" {
+				coderBinary, err = currentBinPath(out)
+				if err != nil {
+					return err
+				}
 			}
```
Comment (Emyrk) [inline cli/configssh.go:245]: I take the input path verbatim. I didn't think I should escape their input.

File: cli/configssh.go [hunk 2]
```diff
@@ -501,6 +508,14 @@ func (r *RootCmd) configSSH() *clibase.Cmd {
 			Description: "Specifies the path to an SSH config.",
 			Value:       clibase.StringOf(&sshConfigFile),
 		},
+		{
+			Flag:    "coder-cli-path",
```
Comment (mafredri) [inline cli/configssh.go:512]: How about just using `binary-path` (without prefix)? I feel like `cli` terminology isn't something we expose the users to in association with the `coder` command, and may cause a disconnect.

File: cli/configssh.go [hunk 3]
```diff
@@ -501,6 +508,14 @@ func (r *RootCmd) configSSH() *clibase.Cmd {
...
+			Env:     "CODER_SSH_CONFIG_CLI_PATH",
+			Default: "",
+			Description: "Optional to specify the path for the coder cli uses in ProxyCommand. " +
+				"By default, the coder cli used is the same cli being invoked with 'config ssh'.",
```
Comment (mafredri) [inline cli/configssh.go:516]: If we change the cli->binary terminology:

```suggestion
			Description: "Optionally specify the path to the coder binary used in ProxyCommand. " +
				"By default, the binary invoking this command ('config ssh') is used.",
```

**Review by mafredri — APPROVED** [review]
Small suggestions, but otherwise looks good.

File: cli/configssh.go (hunk 2 above)
Comment (Emyrk) [reply cli/configssh.go:512]: One thing I was trying to make clear in the flag was this was the `coder` binary path, not anything else like the `ssh` binary. 

Is `coder-binary-path` too long?

**Comment by Emyrk** [comment]
smart to force it to be absolute :+1:. As for who will use it. It is definitely an edge case that I am only adding because someone specifically requested it and it doesn't seem like too much to add.

File: cli/configssh.go (hunk 2 above)
Comment (mafredri) [reply cli/configssh.go:512]: Nah, not too long IMO, your reasoning makes sense 👍🏻

**Pull request merged**